### PR TITLE
use circle-ci 2.0

### PR DESCRIPTION
**Pull request by danielnorberg**
circleci 1.0 will be deprecated end of aug.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          paths:
+            - ~/.m2
+      - run:
+          environment:
+            # Configure datastore emulator heap size
+            JAVA: java -Xmx512m -Xms512m
+            # Maven heap size
+            MAVEN_OPTS: -Xmx256m -Xms256m
+          command: mvn -B verify scala:doc-jar -Pcoverage
+      - run: bash <(curl -s https://codecov.io/bash)
```
Comment (NatashaL) [inline .circleci/config.yml]: If I understood this correctly, running this line will simply download the codecov uploader to `bash`, but it's not actually being used for uploading any coverage reports.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - test_jdk8_with_coverage:
+          filters:
+            branches:
+              ignore: gh-pages
+# FIXME
```
Comment (NatashaL) [inline .circleci/config.yml]: What's the issue to be fixed here?

File: .circleci/config.yml (hunk 1 above)
Comment (danielnorberg) [reply .circleci/config.yml]: It’s calling bash to run the uploader downloaded using curl.

File: .circleci/config.yml (hunk 2 above)
Comment (danielnorberg) [reply .circleci/config.yml]: Too much for this PR. There is another wip PR for that.

**Comment by NatashaL** [comment]
👍

**Pull request merged**